Add climbing presets + fields

zbycz opened the pull request:
Hi,

I'd love to map more climbing routes according [the climbing schema on wiki](https://wiki.openstreetmap.org/wiki/Climbing).
See the discussion in: 
- #31

### This PR
This PR adds the viable minumum to tag climbing crags and climbing routes. I will add more features as needed in future (eg. bouldering, area, alternative tagging for crag and boulder etc). 

I add only presets `climbing/crag` and `climbing/route` + `climbing/route_bottom` for start.

I built this PR with `dist` command and deployed an iD editor with those presets, so it can be tested thorougly.
eg.
- https://osm.zby.cz/id-climbing/#id=r17130664
- https://osm.zby.cz/id-climbing/#id=r14497828


.

.


#### Climbing terminology
The taxonomy goes like this - it can be grouped by relations, or just by nearstanding nodes:
- Climbing area
  - Climbing crag
    - Climbing route
    - Climbing route (start)
    - Climbing route (top)   <-- in climbing terminology "top" is end of route
  - Climbing boulder
    - Climbing route
    - Climbing route (start)
    - Climbing route (top)


### Number of items

| preset name            | tags                                |  | types from wiki    | # 2/24 | # 1/25
|------------------------|-------------------------------------|--|--------------------|-----------| -----
| Climbing area          | climbing = area                     |  | node,area,relation | <1k       | 
| Climbing crag          | climbing = crag                     |  | node,relation      | 3k        | 5k
| ↳ or                   | natural = cliff<br>sport = climbing |  | node,way           | 7k        | 9k
| Climbing boulder       | climbing = boulder                  |  | node,relation      | 1k        |  
| ↳ or                   | natural = stone<br>sport = climbing |  | node,area          | 1k        | 
| Climbing route         | climbing = route                    |  | node,way           | 1k        | 
| Climbing route (start) | climbing = route_bottom             |  | node               | 4k        | 14k
| Climbing route (top)   | climbing = route_top                |  | node               | <1k       |

endolith commented:
Everything should have sport=climbing, no?

zbycz commented:
Thanks @endolith, very good point. I added the `sport=climbing` via the `addTags` property. You can test it in the demo iD editor.

endolith commented:
![image](https://github.com/openstreetmap/id-tagging-schema/assets/58611/8a34bbf8-7dd8-4874-8085-de293c83b7da)

The grade options are always the same?  No YDS in the US?

It complains about some points being on trails or cliffs:

![2024-02-12 22_54_04-iD - osm zby cz_ – Range of Motion — Mozilla Firefox](https://github.com/openstreetmap/id-tagging-schema/assets/58611/2adb75f1-5e5a-44de-98de-dd5159367bcf)
![2024-02-12 22_57_34-iD - osm zby cz_ – Cat in the Hat — Mozilla Firefox](https://github.com/openstreetmap/id-tagging-schema/assets/58611/3a873c9d-fa9e-44f9-bce8-4210c7412ffc)

But not others:

![2024-02-12 22_57_52-iD - osm zby cz_ – Point — Mozilla Firefox](https://github.com/openstreetmap/id-tagging-schema/assets/58611/44a4ded9-36da-4b2b-9e67-2e0ed7ecb9d7)

And doesn't show the same interface for them:

![2024-02-12 22_52_42-iD - osm zby cz_ – LA Roof — Mozilla Firefox](https://github.com/openstreetmap/id-tagging-schema/assets/58611/30eec3d2-45b9-47c5-a104-47adb95041ad)

zbycz commented:
Thanks for the feedback, @endolith 🙂  

ad grades) unfortunately tagging allows for different marks to be assigned for different grade systems. So we have to add a field for each of them in iD. I added the YDS, but in future it would perhaps make sense to create a custom iD field type (like type=multi), which would have better UX (user is not supposed to fill in all of them).
```
climbing:grade:uiaa=*
climbing:grade:french=*
climbing:grade:saxon=*
climbing:grade:yds_class=*
```

ad "It complains about some points being on trails or cliffs") Good point. I added `vertex` type for the `route_bottom` preset. 

ad "But not others") I didn't create a preset for route_top for start, but it is easy now. I added the preset.

Please check it again in the iD demo, eg [Cat in the Hat](https://osm.zby.cz/id-climbing/#id=n4901371127)

tyrasd reviewed (changes requested):
Hi. Thanks for this comprehensive pull request. It looks already quite good. Please find some inline comments, questions and suggestions for some details below.

And as a general remark, the `label` property of fields and the `name` of the presets should follow the so called [_title case_](https://en.wikipedia.org/wiki/Title_case) spelling. For example: it should be `Climbing Route` instead of `Climbing route`. Would you mind adjusting this?

zbycz commented on data/fields/climbing/bolt.json:
```diff
@@ -0,0 +1,17 @@
+    "label": "Bolt type",
+    "strings": {
+        "options": {
+            "expansion": "expansion bolt",
+            "glue-in": "glue-in bolt",
+            "ring": "ring bolt"
```
My mistake. Wiki specifies this tag for `climbing=bolt`, i customized it to fit `climbing=route` as well. 

Removed from this PR, as I don't want to add `climbing=bolt` yet, maybe in a followup PR.

zbycz commented on data/fields/climbing/bolt.json:
```diff
@@ -0,0 +1,17 @@
+{
+    "key": "climbing:bolt",
```
Again the same mistake. 

Removed from this PR, as I don't want to add `climbing=bolt` yet, maybe in a followup PR.

zbycz commented on data/fields/climbing/rock.json:
```diff
@@ -0,0 +1,11 @@
+{
+    "key": "climbing:rock",
+    "type": "combo",
+    "options": [
+        "limestone",
+        "sandstone",
+        "granite",
+        "basalt"
+    ],
```
Well, my aim here was to make all the options translatable. I think this is important, because us non-native speakers usually know the rock types only in our language.

I removed `options` and added proper `strings.options` according to taginfo. Is it ok? Can we somehow suggest that people can add other options as well?

zbycz commented on data/presets/climbing/route.json:
```diff
@@ -0,0 +1,38 @@
+    "fields": [
+        "name",
+        "climbing/length",
+        "climbing/grade/uiaa",
+        "climbing/grade/french",
+        "climbing/grade/saxon",
+        "climbing/grade/yds_class"
```
Thanks, done.

zbycz commented on data/presets/climbing/route_bottom.json:
```diff
@@ -0,0 +1,25 @@
+        "vertex"
+    ],
+    "fields": [
+        "{climbing/route}"
```
I think the issue here is that `climbing=route_bottom` is very often used as a synonym for "here is the route", and in that case it technically suits as `climbing=route`.

Eg. there are 4822 nodes with this tag, and 4575 have name ([overpass](https://overpass-turbo.eu/s/1Jg1)) which may imply this case.

The wiki even specifies this scenario:
> Climbing routes can be mapped with a single node marking the bottom of the route, tagged as climbing=route_bottom, or two nodes tagged with climbing=route_bottom and climbing=route_top connected by a way tagged as climbing=route.

Do you think we can leave the "route" fields on the "route_bottoms" then?

zbycz commented on data/presets/climbing/route_bottom.json:
```diff
@@ -0,0 +1,25 @@
+        "sport": "climbing",
+        "climbing": "route_bottom"
+    },
+    "name": "Climbing route (start)"
```
Done

zbycz commented on data/presets/climbing/route_top.json:
```diff
@@ -0,0 +1,25 @@
+        "vertex"
+    ],
+    "fields": [
+        "{climbing/route}"
+    ],
+    "moreFields": [
+        "{climbing/route}"
```
see above :)

zbycz commented:
@tyrasd Thank you very much for such a good review. Also I am glad I learned how to correctly propose fields, eg. that I should search taginfo first 🙂

Changes deployed to my id instance, eg:
- crag relation: https://osm.zby.cz/id-climbing/#id=r17130664
- crag relation: https://osm.zby.cz/id-climbing/#id=r14497828
- crag line: https://osm.zby.cz/id-climbing/#id=w745451490
- crag area: https://osm.zby.cz/id-climbing/#id=w743598029
- route_bottom without name: https://osm.zby.cz/id-climbing/#id=n2351544812
- route_bottom+route both with names: https://osm.zby.cz/id-climbing/#id=n4901371127 (Cat in a Hat)


While editing moreFields I also got an idea how to make them more usable: https://github.com/openstreetmap/iD/pull/10181

zbycz commented:
Hi @tyrasd - just gently bumping this issue. 🙂 🙏

zbycz commented:
Hallo @tyrasd & @tordans 👋 

I have an update to this PR after one year. 😅  Meanwhile the climbing project got bigger. We have now a running site [openclimbing.org](https://openclimbing.org/) which allows easy browsing and creating of climbing features (and uses id-tagging-schema as well). This means, that now we have 14k nodes of `climbing=route_bottom` and 5k relations of `climbing=crag`.

I went through this whole PR and it still makes sense, waiting only for two last responses in the @tyrasd's code review (see the unresolved comments in the review above). If we find the answers I think we could move toward merging it and make climbing even better 💪 🙂 

I updated the table in PR description with counts, here are only the presets proposed in this PR:



| preset name          | tags                    | types in PR              | # in 2/24 | # now 
|----------------------|-------------------------|--------------------------|-----------|-------|
| Climbing Crag        | climbing = crag         | point,line,area,relation | 3k        | **5k**    
| Climbing Route       | climbing = route        | point,line               | 1k        |   
| Climbing Route Start | climbing = route_bottom | point,vertex             | 4k        | **14k**   
| Climbing Route Top   | climbing = route_top    | point,vertex             | <1k       |

zbycz commented:
Closing this huge PR in favor of smaller one: https://github.com/openstreetmap/id-tagging-schema/pull/1590
